server/opts.go

```diff
@@ -2746,6 +2750,16 @@ func parseRemoteLeafNodes(v any, errors *[]error, warnings *[]error) ([]*RemoteL
 				remote.Websocket.NoMasking = v.(bool)
 			case "jetstream_cluster_migrate", "js_cluster_migrate":
 				remote.JetStreamClusterMigrate = true
+				migrateConfig, ok := v.(map[string]any)
```

Comment:
Although it was this way before, currently setting `jetstream_cluster_migrate: false` in the config looks like it will do the wrong thing and set to `true` anyway. Do we want to fix this?

Might also be better just to check for bool vs string, parse the string as duration, rather than having to open up a new map. That way you could do any of:
```
jetstream_cluster_migrate: true
jetstream_cluster_migrate: 30s
jetstream_cluster_migrate: false
```
WDYT?

Reply:
Maybe @derekcollison has an opinion on this too.

Reply:
I would prefer `leader_migrate_delay`, although that is pretty long winded for me but is more descriptive IMO. It is a value that defaults to 0, meaning instant, and you can set to a value that means after Nsecs of being disconnected from leafnode we will migrate leaders.

Reply:
@neilalexander I agree but I wonder how much of a breaking change that would be `js_cluster_enabled: false` seems like a bug which can be fixed, but right now e.g. `js_cluster_migrate: enabled` also works (or any other value really).

@derekcollison with `leader_migrate_delay` you would see this as a separate option in `remote` or an option in object on `js_cluster_migrate` (as I did it here with `delay`)?

Reply:
option on js_cluster_migrate I think.

Reply:
Let's make sure `js_cluster_migrate: false` works correctly.